aya/src/maps/task_storage.rs

```diff
@@ -0,0 +1,154 @@
+//! Task storage.
+use std::{
+    borrow::Borrow,
+    ffi::c_int,
+    marker::PhantomData,
+    os::fd::{AsFd, AsRawFd},
+};
+
+use crate::{
+    maps::{check_kv_size, MapData, MapError},
+    sys::{bpf_map_lookup_elem, PidFd, SyscallError},
+    Pod,
+};
+
+/// Task storage is a type of map which uses `task_struct` kernel type as a
+/// key. When the task (process) stops, the corresponding entry is
+/// automatically removed.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 5.12.
+///
+/// # Examples
+///
+/// ```no_run
+/// # let mut ebpf = aya::Ebpf::load(&[])?;
+/// use aya::maps::TaskStorage;
+///
+/// let mut task_storage: TaskStorage<_, u32> = TaskStorage::try_from(ebpf.map_mut("TASK_STORAGE").unwrap())?;
+///
+/// let pid = 0;
+/// let value = task_storage.get(&pid, 0)?;
+/// # Ok::<(), aya::EbpfError>(())
+/// ```
+#[doc(alias = "BPF_MAP_TYPE_TASK_STORAGE")]
+#[derive(Debug)]
+pub struct TaskStorage<T, V> {
+    pub(crate) inner: T,
+    _v: PhantomData<V>,
+}
+
+impl<T: Borrow<MapData>, V: Pod> TaskStorage<T, V> {
+    pub(crate) fn new(map: T) -> Result<Self, MapError> {
+        let data = map.borrow();
+        check_kv_size::<c_int, V>(data)?;
```

Comment:
Other map types use:
```suggestion
        check_kv_size::<u32, V>(data)?;
```

If you can't use u32 - although I'm not sure why not - a comment to explain would be helpful.